src/main/java/uk/gov/gchq/magmacore/service/DbCreateOperation.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2021 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package uk.gov.gchq.magmacore.service;
+
+import java.util.function.Function;
+
+import uk.gov.gchq.hqdm.model.Thing;
+import uk.gov.gchq.hqdm.rdf.exception.HqdmException;
+import uk.gov.gchq.hqdm.rdf.iri.IRI;
+import uk.gov.gchq.hqdm.services.SpatioTemporalExtentServices;
+import uk.gov.gchq.magmacore.exception.DbTransformationException;
+
+/**
+ * Class representing an invertible operation to create a predicate.
+ */
+public class DbCreateOperation implements Function<MagmaCoreService, MagmaCoreService> {
+
+    // The IRI of the Thing we're referring to.
+    private IRI subject;
+
+    // The IRI of the property we're referring to.
+    private IRI predicate;
+
+    // The value of the property we're referring to.
+    private String object;
+
+    /**
+     * Constructor.
+     *
+     * @param subject {@link IRI}
+     * @param predicate {@link IRI}
+     * @param object {@link String}
+    */
+    public DbCreateOperation(final IRI subject, final IRI predicate, final String object) {
+        this.subject = subject;
+        this.predicate = predicate;
+        this.object = object;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * */
+    @Override
+    public MagmaCoreService apply(final MagmaCoreService mcService) {
+        Thing thing = null;
+        try {
+            thing = mcService.get(subject);
+        } catch (final HqdmException e) {
```

Comment:
The exception is handled by the subsequent code. This situation arises because one implementation of `MagmaCoreDatabase` returns `null` if the object does not exist and another throws an exception, so this code turns both into `null means not already existing` so it can be handled by the subsequent code in the same way. I think there should be a bug raised to make all implementations of `MagmaCoreInterface` behave the same - either return `null` or throw the same exception in all cases.

Reply:
Agreed, I didn't realise there was a mismatch between implementations of `MagmaCoreDatabase`. Is that something you could fix as part of these changes or would you prefer to raise a bug report for it?

Reply:
The try/catch seems no longer necessary now that the `Object` database has gone, so I've removed it.